### PR TITLE
simplify interactions with arrow flight APIs

### LINKED ISSUE #376: simplify interactions with arrow flight APIs
**Is your feature request related to a problem or challenge? Please describe what you are trying to do.**
The current APIs are functional and with some source code digging, you can use them to accomplish your goals. However, I think it would be fair to note that they are not so convenient or easy to use as the comparable Python (or other language) implementations. Particularly on the client side.

**Describe the solution you'd like**
Extend the existing basic implementation and make it easier for developers to work with Flight in rust. Some ideas include:
- constructors for important types
- support for useful traits (e.g.: From, Display) for important types
- restrict access to "untyped" data (e.g.: Vec<u8>) and provide more typing

**Describe alternatives you've considered**
Look at the existing C++/Java/Python implementations for inspiration

**Additional context**
It should be as simple to create Flight client/servers in rust as it is in C++

### PR DESCRIPTION
**Pull request by garyanaplan**
Initial work to implement some basic traits and illustrate the direction of travel

# Which issue does this PR close?

closes #376 

# Rationale for this change
 
Illustrating direction of travel. Open Questions:

 - ~~Should I put this into a separate module or is it ok in utils.rs?~~
 - I've moved the trait stuff into src/lib.rs and hidden the generated code into a `gen` module.
 - Would it be better to replace Vec<u8> with NewTypes (such as IpcMessage that I added to the src/utils.rs)
 - ~~Should I work on hiding some of the raw generated code?~~
 - I've hidden the generated code and re-exported the documented bits that I think are required.

# What changes are included in this PR?

Illustrates what kind of work I'm doing before I go too far in the wrong direction

# Are there any user-facing changes?

Deprecation warnings...

### CODE REVIEW

**Review by nevi-me — COMMENTED** [review]
Just a few comments @garyanaplan

File: arrow-flight/src/utils.rs [hunk 1]
```diff
@@ -164,4 +73,40 @@ pub fn flight_data_to_arrow_batch(
         })?
 }
 
-// TODO: add more explicit conversion that exposes flight descriptor and metadata options
+/// Convert a `Schema` to `SchemaResult` by converting to an IPC message
+#[deprecated(
+    since = "4.3.0",
```
[the remark this answers is not in the document]
Comment (garyanaplan) [reply arrow-flight/src/utils.rs]: Done

File: arrow-flight/src/lib.rs [hunk 2]
```diff
@@ -15,6 +15,441 @@
...
+}
+
+// TryFrom...
+
+impl TryFrom<i32> for DescriptorType {
+    type Error = ArrowError;
+
+    fn try_from(value: i32) -> ArrowResult<Self> {
+        match value {
+            0 => Ok(DescriptorType::Unknown),
```
[the remark this answers is not in the document]
Comment (garyanaplan) [reply arrow-flight/src/lib.rs]: Done

**Review by nevi-me — APPROVED** [review]

**Comment by nevi-me** [comment]
@alamb I've marked this as an API change because the import paths for the proto module have changed.

**Pull request merged**